src/wallet/mod.rs

```diff
@@ -196,24 +197,52 @@ pub enum AddressIndex {
     Reset(u32),
 }
 
+/// A derived address and the index it was found at
+/// For convenience this automatically derefs to `Address`
+#[derive(Debug, PartialEq)]
+pub struct AddressInfo {
+    /// Child index of this address
+    pub index: u32,
+    /// Address
+    pub address: Address,
+}
+
+impl Deref for AddressInfo {
+    type Target = Address;
+
+    fn deref(&self) -> &Self::Target {
+        &self.address
+    }
+}
+
+impl fmt::Display for AddressInfo {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.address)
+    }
+}
+
 // offline actions, always available
 impl<B, D> Wallet<B, D>
 where
     D: BatchDatabase,
 {
     // Return a newly derived address using the external descriptor
-    fn get_new_address(&self) -> Result<Address, Error> {
+    fn get_new_address(&self) -> Result<(Address, u32), Error> {
         let incremented_index = self.fetch_and_increment_index(KeychainKind::External)?;
 
-        self.descriptor
+        let address_result = self
+            .descriptor
             .as_derived(incremented_index, &self.secp)
-            .address(self.network)
+            .address(self.network);
+
+        address_result
+            .map(|address| (address, incremented_index))
```

Comment:
I just realized I don't have to return a tuple here, I could return `AddressInfo` as well (in all four of these private methods). Thoughts on what would be preferable?